frontends/p4/unusedDeclarations.cpp

```diff
@@ -112,7 +112,7 @@ const IR::Node *RemoveUnusedDeclarations::process(const IR::IDeclaration *decl)
     LOG3("Visiting " << decl);
     if (decl->getName().name == IR::ParserState::verify && getParent<IR::P4Program>())
         return decl->getNode();
-    if (decl->getName().name.startsWith("__"))
+    if (decl->externalName(decl->getName().name).startsWith("__"))
```

Comment:
What happens when you have something like `@name("a2") action _xyz()` now?

Reply:
@fruffy Please see the added test outputs

Reply:
tagging @oleg-ran-amd for any additional comments